src/services/legacyreports.js

```diff
@@ -0,0 +1,112 @@
+import { Op } from 'sequelize';
+import { userByEmail } from './users';
+import { ActivityReport, ActivityReportCollaborator } from '../models';
+import { logger } from '../logger';
+
+const getLegacyReports = async () => {
+  const reports = await ActivityReport.findAll({
+    where: {
+      legacyId: {
+        [Op.ne]: null,
+      },
+      imported: {
+        [Op.ne]: null,
+      },
+      [Op.or]: [
+        {
+          userId: {
+            [Op.eq]: null,
+          },
+        },
+        {
+          approvingManagerId: {
+            [Op.eq]: null,
+          },
+        },
+        {
+          imported: {
+            otherSpecialists: {
+              [Op.ne]: '',
+            },
+          },
+        },
+      ],
+
+    },
+  });
+  return reports;
+};
+
+export const reconcileApprovingManagers = async (report) => {
+  try {
+    const user = await userByEmail(report.imported.manager);
+    if (user) {
+      await ActivityReport.update({ approvingManagerId: user.id }, { where: { id: report.id } });
+      logger.info(`Updated approvingManager for report ${report.displayId} to user Id ${user.id}`);
+    }
+  } catch (err) {
+    logger.error(err);
+  }
+};
+
+export const reconcileAuthors = async (report) => {
+  try {
+    const user = await userByEmail(report.imported.createdBy);
+    if (user) {
+      await ActivityReport.update({ userId: user.id }, { where: { id: report.id } });
+      logger.info(`Updated author for report ${report.displayId} to user Id ${user.id}`);
+    }
+  } catch (err) {
+    logger.error(err);
+  }
+};
+
+export const reconcileCollaborators = async (report) => {
+  try {
+    const collaborators = await ActivityReportCollaborator
+      .findAll({ where: { activityReportId: report.id } });
+    const otherSpecialists = report.imported.otherSpecialists.split(',').filter((j) => j !== '').map((i) => i.toLowerCase().trim());
+    if (otherSpecialists.length !== collaborators.length) {
```

Comment:
no, but since I am using find or create to add the collaborators duplicates shouldn't matter right?

Reply:
I was more thinking about this "if" statement running or not running since we compare the length. However after thinking some more we should be okay. If there are smartsheet duplicates the "if" will run after a report has all collaborators but as you say you use `find or create` so it shouldn't really matter.